bip-0159.mediawiki

```diff
@@ -42,13 +44,15 @@ Full nodes following this BIP <I>SHOULD</I> relay address/services (<code>addr</
 
 === Counter-measures for peer fingerprinting ===
 
-Peers may have different prune depths (depending on the peers configuration, disk space, etc.) which can result in a fingerprinting weakness (finding the prune depth through getdata requests). NODE_NETWORK_LIMITED supporting peers <I>SHOULD</I> avoid leaking the prune depth and therefore not serve blocks deeper than the signaled <code>NODE_NETWORK_LIMITED</code> threshold (288 blocks).
+Peers may have different prune depths (depending on the peers configuration, disk space, etc.), which can result in a fingerprinting weakness (finding the prune depth through getdata requests).
+
+Pruned peers (<code>NODE_NETWORK_LIMITED</code> nodes that do not also signal serving the full block chain) should avoid leaking the prune depth and therefore <I>SHOULD NOT</I> serve blocks deeper than the signaled <code>NODE_NETWORK_LIMITED</code> threshold of 288 blocks.
```

Comment:
This seems unnecessarily specific. A node that signals `NODE_NETWORK_LIMITED` but not `NODE_NETWORK` should serve only the last 288 blocks, regardless of whether they are pruned or not.

```suggestion
Peers that signal <code>NODE_NETWORK_LIMITED</code> that do not also signal <code>NODE_NETWORK</code> should avoid leaking the prune depth and therefore <I>SHOULD NOT</I> serve blocks deeper than the signaled <code>NODE_NETWORK_LIMITED</code> threshold of 288 blocks.
```

Reply:
Thanks, done.